tests/infer/test_gradient.py

```diff
@@ -73,31 +74,30 @@ def test_iarange(Elbo, reparameterized):
     Normal = dist.Normal if reparameterized else fakes.NonreparameterizedNormal
 
     def model():
-        x = data.unsqueeze(-1).expand(-1, num_particles)
-        data_iarange = pyro.iarange("data", len(data), dim=-2)
-        particles_iarange = pyro.iarange("particles", num_particles, dim=-1)
+        particles_iarange = pyro.iarange("particles", num_particles, dim=-2)
+        data_iarange = pyro.iarange("data", len(data), dim=-1)
 
         pyro.sample("nuisance_a", Normal(0, 1))
         with particles_iarange, data_iarange:
-            z = pyro.sample("z", Normal(0, 1).expand_by(x.shape))
+            z = pyro.sample("z", Normal(0, 1))
         pyro.sample("nuisance_b", Normal(2, 3))
         with data_iarange, particles_iarange:
-            pyro.sample("x", Normal(z, 1), obs=x)
+            pyro.sample("x", Normal(z, 1), obs=data)
         pyro.sample("nuisance_c", Normal(4, 5))
 
     def guide():
-        loc = pyro.param("loc", lambda: torch.zeros(len(data), requires_grad=True))
-        scale = pyro.param("scale", lambda: torch.tensor([1.0], requires_grad=True))
-        mus = loc.unsqueeze(-1).expand(-1, num_particles)
+        loc = pyro.param("loc", torch.zeros(len(data)))
+        scale = pyro.param("scale", torch.tensor([1.]))
 
         pyro.sample("nuisance_c", Normal(4, 5))
-        with pyro.iarange("particles", num_particles):
-            with pyro.iarange("data", len(data)):
-                pyro.sample("z", Normal(mus, scale))
+        with pyro.iarange("particles", num_particles, dim=-2):
+            with pyro.iarange("data", len(data), dim=-1):
+                pyro.sample("z", Normal(loc, scale))
         pyro.sample("nuisance_b", Normal(2, 3))
         pyro.sample("nuisance_a", Normal(0, 1))
 
     optim = Adam({"lr": 0.1})
+    model, guide = poutine.broadcast(model), poutine.broadcast(guide)
```

Comment:
nit: I think it's a little clearer to decorate
```py
@poutine.broadcast
def model():
    ...

@poutine.broadcast
def guide():
    ...
```
That way readers know as soon as they start reading the model that it should be read with broadcast semantics.